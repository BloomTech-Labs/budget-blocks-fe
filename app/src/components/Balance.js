import React from "react";
import LinkAccount from "./LinkAccount";
import plaidImg from "../media/image/PlaidIcon.png";
import { connect } from "react-redux";

export const Balance = (props)=>{
    if(props.LinkedAccount ===true ){
        return(
            <div className="Balance">
                <div className="myBalance">
                    <p>My Balance</p>
                    <h3>{`$${props.balance}`}</h3>
                </div>
            </div>
        )
    }else{
        return(
            <div className="NoBalance">
                <LinkAccount/>
                <img src={plaidImg} className="plaidIcon" alt="Plaid Icon"/>
            </div>
        )
    }
}

function mapStateToProps(state){
    return {
        balance:state.plaidReducer.Balance,
        LinkedAccount: state.loginReducer.user.LinkedAccount
    }
}

export default connect(mapStateToProps,{})(Balance)