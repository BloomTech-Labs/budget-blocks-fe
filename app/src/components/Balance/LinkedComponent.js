import React from "react";
import { connect } from "react-redux";

const LinkedComponent = (props)=>{

    return(
        <div className="Balance">
            <div className="myBalance">
                <p>My Balance</p>
                <h3>{`$${props.balance}`}</h3>
            </div>
        </div>
    )
}

function mapStateToProps(state){
    return {
        balance:state.plaidReducer.Balance
    }
}

export default connect(mapStateToProps,{})(LinkedComponent)