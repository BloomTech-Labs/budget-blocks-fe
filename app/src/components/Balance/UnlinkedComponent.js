import React from "react";
import LinkAccount from "../LinkAccount";
import plaidImg from "../../media/image/PlaidIcon.png";
const UnlinkedComponent = (props)=>{

    return(
        <div className="NoBalance">
            <LinkAccount/>
            <img src={plaidImg} className="plaidIcon" alt="Plaid Icon"/>
        </div>
    )
}

export default UnlinkedComponent