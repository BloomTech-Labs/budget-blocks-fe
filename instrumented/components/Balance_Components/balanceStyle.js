var cov_p9r0s6v28 = function () {
  var path = "/Users/Cha/Desktop/Lambda/budget-blocks-fe/src/components/Balance_Components/balanceStyle.js";
  var hash = "2db0fad8ec8a216ba4ff901a686a21804f8a1777";
  var global = new Function("return this")();
  var gcv = "__coverage__";
  var coverageData = {
    path: "/Users/Cha/Desktop/Lambda/budget-blocks-fe/src/components/Balance_Components/balanceStyle.js",
    statementMap: {
      "0": {
        start: {
          line: 3,
          column: 25
        },
        end: {
          line: 14,
          column: 2
        }
      }
    },
    fnMap: {},
    branchMap: {},
    s: {
      "0": 0
    },
    f: {},
    b: {},
    _coverageSchema: "43e27e138ebf9cfc5966b082cf9a028302ed4184",
    hash: "2db0fad8ec8a216ba4ff901a686a21804f8a1777"
  };
  var coverage = global[gcv] || (global[gcv] = {});

  if (coverage[path] && coverage[path].hash === hash) {
    return coverage[path];
  }

  return coverage[path] = coverageData;
}();

import { makeStyles } from '@material-ui/core/styles';
export const useStyles = (cov_p9r0s6v28.s[0]++, makeStyles({
  NoBalance: {
    minWidth: 275,
    background: '#F0F0F0',
    display: "flex",
    padding: "2rem",
    marginBottom: "2rem",
    marginTop: "2rem",
    alignItems: "center",
    justifyContent: "space-around"
  }
}));