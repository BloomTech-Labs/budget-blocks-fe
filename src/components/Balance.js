import React,{useEffect} from "react";
import { connect } from "react-redux";
import {getUserInfo} from '../redux/actions/ProfileActions';
import LinkAccount from "./LinkAccount";
import plaidImg from "../media/image/PlaidIcon.png";

export const Balance = ({balance, LinkedAccount})=>{
    useEffect(()=> {
        getUserInfo()
    },[])
    const amounts = balance.accounts.map(e => e.balances.available)
    const sum = amounts.reduce((a,b) => {return a + b},0)

    if(LinkedAccount === false){
        return(
        <div className="NoBalance">
            <LinkAccount/>
            <img src={plaidImg} className="plaidIcon" alt="Plaid Icon"/>
        </div>
        )
    }else{
        return (
        <div className="Balance">
            <h1>
                ${sum}
            </h1>
        </div>
             
        )
    }
}

function mapStateToProps(state){
    return {
        balance:state.plaidReducer,
        LinkedAccount: state.loginReducer.user.LinkedAccount
    }
}

export default connect(mapStateToProps,{getUserInfo})(Balance)