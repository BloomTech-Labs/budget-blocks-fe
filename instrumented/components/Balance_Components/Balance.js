var cov_o8fridqe8 = function () {
  var path = "/Users/Cha/Desktop/Lambda/budget-blocks-fe/src/components/Balance_Components/Balance.js";
  var hash = "d7d9347f2a028cc0b24f606b53e5bd846ade3ff6";
  var global = new Function("return this")();
  var gcv = "__coverage__";
  var coverageData = {
    path: "/Users/Cha/Desktop/Lambda/budget-blocks-fe/src/components/Balance_Components/Balance.js",
    statementMap: {
      "0": {
        start: {
          line: 8,
          column: 23
        },
        end: {
          line: 30,
          column: 1
        }
      },
      "1": {
        start: {
          line: 13,
          column: 20
        },
        end: {
          line: 13,
          column: 31
        }
      },
      "2": {
        start: {
          line: 16,
          column: 4
        },
        end: {
          line: 29,
          column: 5
        }
      },
      "3": {
        start: {
          line: 18,
          column: 8
        },
        end: {
          line: 23,
          column: 9
        }
      },
      "4": {
        start: {
          line: 26,
          column: 8
        },
        end: {
          line: 28,
          column: 9
        }
      },
      "5": {
        start: {
          line: 33,
          column: 1
        },
        end: {
          line: 35,
          column: 3
        }
      }
    },
    fnMap: {
      "0": {
        name: "(anonymous_0)",
        decl: {
          start: {
            line: 8,
            column: 23
          },
          end: {
            line: 8,
            column: 24
          }
        },
        loc: {
          start: {
            line: 8,
            column: 46
          },
          end: {
            line: 30,
            column: 1
          }
        },
        line: 8
      },
      "1": {
        name: "mapStateToProps",
        decl: {
          start: {
            line: 32,
            column: 9
          },
          end: {
            line: 32,
            column: 24
          }
        },
        loc: {
          start: {
            line: 32,
            column: 32
          },
          end: {
            line: 36,
            column: 1
          }
        },
        line: 32
      }
    },
    branchMap: {
      "0": {
        loc: {
          start: {
            line: 16,
            column: 4
          },
          end: {
            line: 29,
            column: 5
          }
        },
        type: "if",
        locations: [{
          start: {
            line: 16,
            column: 4
          },
          end: {
            line: 29,
            column: 5
          }
        }, {
          start: {
            line: 16,
            column: 4
          },
          end: {
            line: 29,
            column: 5
          }
        }],
        line: 16
      }
    },
    s: {
      "0": 0,
      "1": 0,
      "2": 0,
      "3": 0,
      "4": 0,
      "5": 0
    },
    f: {
      "0": 0,
      "1": 0
    },
    b: {
      "0": [0, 0]
    },
    _coverageSchema: "43e27e138ebf9cfc5966b082cf9a028302ed4184",
    hash: "d7d9347f2a028cc0b24f606b53e5bd846ade3ff6"
  };
  var coverage = global[gcv] || (global[gcv] = {});

  if (coverage[path] && coverage[path].hash === hash) {
    return coverage[path];
  }

  return coverage[path] = coverageData;
}();

import React from 'react';
import { connect } from 'react-redux';
import LinkAccount from '../LinkAccount';
import plaidImg from '../../media/image/PlaidIcon.png';
import { useStyles } from './balanceStyle';
import Card from '@material-ui/core/Card';
cov_o8fridqe8.s[0]++;
export const Balance = ({
  LinkedAccount
}) => {
  cov_o8fridqe8.f[0]++;
  //This component was originally meant to be used on the dashboard. If the user had a manual account and wanted to swith to using plaid: this component would allow that
  //In the current version, you cannot have a mix of plaid and manual so this component is only used in the onboarding process
  //uses styling from the balanceStlye file. 
  const classes = (cov_o8fridqe8.s[1]++, useStyles());
  cov_o8fridqe8.s[2]++;

  if (LinkedAccount === false) {
    cov_o8fridqe8.b[0][0]++;
    cov_o8fridqe8.s[3]++;
    //if the user's account does not use plaid then it displays component to connect bank account
    return <Card className={classes.NoBalance}>
            <LinkAccount />
            <img src={plaidImg} className="plaidIcon" alt="Plaid Icon" />
        </Card>;
  } else {
    cov_o8fridqe8.b[0][1]++;
    cov_o8fridqe8.s[4]++;
    //if the user has already connected their bank account then display nothing
    return <p></p>;
  }
};

function mapStateToProps(state) {
  cov_o8fridqe8.f[1]++;
  cov_o8fridqe8.s[5]++;
  return {
    LinkedAccount: state.loginReducer.user.LinkedAccount
  };
}

export default connect(mapStateToProps, {})(Balance);