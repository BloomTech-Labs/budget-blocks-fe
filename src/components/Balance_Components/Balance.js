import React from "react";
import { connect } from "react-redux";
import LinkAccount from "../LinkAccount";
import plaidImg from "../../media/image/PlaidIcon.png";
import { useStyles } from "./balanceStyle";
import Card from "@material-ui/core/Card";
import environmentUrls from "../../dispatch";

export const Balance = ({ LinkedAccount }) => {
  //This component was originally meant to be used on the dashboard. If the user had a manual account and wanted to switch to using plaid: this component would allow that
  //In the current version, you cannot have a mix of plaid and manual so this component is only used in the onboarding process

  //uses styling from the balanceStlye file.
  const classes = useStyles();

  if (LinkedAccount === false) {
    //if the user's account does not use plaid then it displays component to connect bank account
    return (
      <Card className={classes.NoBalance}>
        <LinkAccount />
        <img src={plaidImg} className="plaidIcon" alt="Plaid Icon" />
      </Card>
    );
  } else {
    //if the user has already connected their bank account then display nothing
    return <p></p>;
  }
};

function mapStateToProps(state) {
  return {
    LinkedAccount: state.loginReducer.user.LinkedAccount
  };
}

export default connect(mapStateToProps, {})(Balance);
