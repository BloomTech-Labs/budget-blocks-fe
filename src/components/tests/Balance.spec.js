import React from "react";
import  { Balance }  from "../Balance_Components/Balance";
import { render, fireEvent } from "@testing-library/react";
import {BrowserRouter as Router } from "react-router-dom"
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, render as enzymeRender } from 'enzyme';

configure({ adapter: new Adapter() });

test('Balance renders correctly',()=>{
    const wrapper = shallow(<Balance LinkedAccount={false} balance={{accounts:[]}}/>)
        
    expect(wrapper).toMatchSnapshot();
});

test('Balance renders Link account component when Linked Account is false',()=>{
    const wrapper = shallow(<Balance LinkedAccount={false} />)
        
    const component = wrapper.find("div").at(0);
    expect(component.props().className).toBe("NoBalance");
});

// test('Balance renders Balance when Linked Account is true',()=>{
//     const wrapper = shallow(<Balance LinkedAccount={true} />)
        
//     const component = wrapper.find("div").at(0);
//     expect(component.props().className).toBe("Balance");
// });

