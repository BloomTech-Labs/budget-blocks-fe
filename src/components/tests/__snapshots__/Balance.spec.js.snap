// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balance renders correctly 1`] = `ShallowWrapper {}`;
